src/formatFieldValue.ts

```diff
@@ -9,30 +10,41 @@ import {timeFormat} from 'd3-time-format';
  * @param format - a time time format specifier, or a time number format specifier, or undefined
  * @return the formatted value, or undefined if value or formatType is missing
  */
-export function customFormat(value: number | string | Date, formatType: string, format: string): string {
+export function customFormat(value: ScenegraphPrimitive, formatType: string, format: string | FormatCallback): string | undefined {
   if (value === undefined || value === null) {
     return undefined;
   }
   if (!formatType) {
     return undefined;
   }
 
-  switch (formatType) {
-    case 'time':
-      return format ? timeFormat(format)(value as Date) : autoTimeFormat(value as Date);
-    case 'number':
-      return format ? d3NumberFormat(format)(value as number) : autoNumberFormat(value as number);
-    case 'string':
-    default:
-      return value as string;
+  if (typeof format === 'string') {
+    switch (formatType) {
+      case 'time':
+        return format ? timeFormat(format)(value as Date) : autoTimeFormat(value as Date);
+      case 'number':
+        return format ? d3NumberFormat(format)(value as number) : autoNumberFormat(value as number);
+      case 'string':
+      default:
+        return value as string;
+    }
   }
+
+  if (formatType === 'custom') {
```

Comment:
I wonder if we should just say when `format` is a function, formatType is implicitly always "custom" -- thus we don't have to check it here  and users do not have to explicitly says `formatType: "custom"` and only specify `format`.   That's probably more concise?

Reply:
Good idea! Will do.